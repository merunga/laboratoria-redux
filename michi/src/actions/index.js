export const actionTypes = {
  SELECT: 'SELECT',
  RESET: 'RESET',
}

export const select = (x, y, turno) => ({
  type: actionTypes.SELECT,
  x,
  y,
  turno,
})

export const reset = () => ({
  type: actionTypes.RESET,
})
