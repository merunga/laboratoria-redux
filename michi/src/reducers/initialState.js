export default {
  michi: [
    [null, null, null],
    [null, null, null],
    [null, null, null],
  ],
  // historia: {
  //   comienza: 'X',
  //   jugadas: [],
  // },
}
