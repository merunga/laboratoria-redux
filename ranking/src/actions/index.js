export const actionTypes = {
  UP_VOTE: 'UP_VOTE',
  DOWN_VOTE: 'DOWN_VOTE',
  PREV: 'PREV',
  NEXT: 'NEXT',
}

export const upVote = id => ({
  type: actionTypes.UP_VOTE,
  id,
})

export const downVote = id => ({
  type: actionTypes.DOWN_VOTE,
  id,
})

export const prev = () => ({
  type: actionTypes.PREV,
})

export const next = () => ({
  type: actionTypes.NEXT,
})
