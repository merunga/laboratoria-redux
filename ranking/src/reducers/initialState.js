export default {
  canciones: [
    {
      id: 1, titulo: 'Psycho', autor: 'Post Malone', votos: 70,
    },
    {
      id: 2, titulo: 'Perfect', autor: 'Ed Sheeran', votos: 90,
    },
    {
      id: 3, titulo: 'God\'s Plan', autor: 'Drake', votos: 100,
    },
    {
      id: 4, titulo: 'Finesse', autor: 'Bruno Mars', votos: 80,
    },
    {
      id: 5, titulo: 'Meant To Be', autor: 'Bebe Rexha & Florida Georgia Line', votos: 60,
    },
  ],
  seleccionado: 1,
}
