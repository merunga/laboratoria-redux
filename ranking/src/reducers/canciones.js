import { actionTypes } from '../actions'

export default (state, action) => {
  if (action.type === actionTypes.UP_VOTE) {
    return state.map(c => (c.id === action.id ? { ...c, votos: c.votos + 1 } : c))
  }
  if (action.type === actionTypes.DOWN_VOTE) {
    return state.map(c => (
      c.id === action.id ? { ...c, votos: (c.votos > 0 ? c.votos - 1 : 0) } : c
    ))
  }
  return state || null
}
