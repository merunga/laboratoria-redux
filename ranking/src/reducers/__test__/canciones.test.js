import initialState from '../initialState'
import canciones from '../canciones'
import * as actions from '../../actions'

describe('canciones', () => {
  it('upvotea', () => {
    expect(canciones(initialState.canciones, actions.upVote(1))).toMatchSnapshot()
  })
  it('downvotea', () => {
    expect(canciones(initialState.canciones, actions.downVote(5))).toMatchSnapshot()
  })
})
