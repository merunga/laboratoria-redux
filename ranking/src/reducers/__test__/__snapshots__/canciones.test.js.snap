// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`canciones downvotea 1`] = `
Array [
  Object {
    "autor": "Post Malone",
    "id": 1,
    "titulo": "Psycho",
    "votos": 70,
  },
  Object {
    "autor": "Ed Sheeran",
    "id": 2,
    "titulo": "Perfect",
    "votos": 90,
  },
  Object {
    "autor": "Drake",
    "id": 3,
    "titulo": "God's Plan",
    "votos": 100,
  },
  Object {
    "autor": "Bruno Mars",
    "id": 4,
    "titulo": "Finesse",
    "votos": 80,
  },
  Object {
    "autor": "Bebe Rexha & Florida Georgia Line",
    "id": 5,
    "titulo": "Meant To Be",
    "votos": 59,
  },
]
`;

exports[`canciones upvotea 1`] = `
Array [
  Object {
    "autor": "Post Malone",
    "id": 1,
    "titulo": "Psycho",
    "votos": 71,
  },
  Object {
    "autor": "Ed Sheeran",
    "id": 2,
    "titulo": "Perfect",
    "votos": 90,
  },
  Object {
    "autor": "Drake",
    "id": 3,
    "titulo": "God's Plan",
    "votos": 100,
  },
  Object {
    "autor": "Bruno Mars",
    "id": 4,
    "titulo": "Finesse",
    "votos": 80,
  },
  Object {
    "autor": "Bebe Rexha & Florida Georgia Line",
    "id": 5,
    "titulo": "Meant To Be",
    "votos": 60,
  },
]
`;
